### PR TITLE
Add quickstart example for task queue functions.

### PR DESCRIPTION
**Pull request by taeold**
NOTE: Requires custom build of the admin SDK.

### CODE REVIEW

File: quickstarts/taskqueues-backup-images/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Quickstart: Back up images using a Task Queue Function
```
Comment (egilmorez) [inline quickstarts/taskqueues-backup-images/README.md:1]: Do you think the README should discuss the values chosen for rate limits, backoff, etc?  

Not necessary afaik, but I feel like I'd like to discuss them in the doc/guide.

**Review by egilmorez — COMMENTED** [review]
Thanks!  A few things to look at, and some questions.

File: quickstarts/taskqueues-backup-images/README.md (hunk 1 above)
Comment (taeold) [reply quickstarts/taskqueues-backup-images/README.md:1]: Great suggestion! I'll try to expand on the task queue configuration on the README.md.

File: quickstarts/taskqueues-backup-images/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+This quickstart demonstrates how to setup a Task Queue Function using the Firebase SDK for Cloud Functions.
+
+## Introduction
+
+Task queue functions make it easy to manage the execution, dispatch, and delivery of a large number of distributed tasks.
+
+We leverage its power here to set up a service that backs up all images from NASA's [Astronomy Picture of the Day](https://apod.nasa.gov/apod/astropix.html).
+
+Task queue functions are powered by [Google Cloud Tasks](https://cloud.google.com/tasks). Learn more about the task queue functions at TODO.
```
[the remark this answers is not in the document]
Comment (taeold) [reply quickstarts/taskqueues-backup-images/README.md:10]: Yep that's what I'm hoping to include here. Do we know roughly where that would be?

File: quickstarts/taskqueues-backup-images/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+## Functions
+The sample code consists of 2 functions:
+
+### 1. `backupApod`
+A task queue function responsible for processing the logic for backing up the Astronomy Picture of the Day ("apod") for the given date. This function will be triggered for every task enqueued on the corresponding queue created in Cloud Tasks.
+
+Task queue functions come with a powerful set of configuration to precisely control rate limits and retry behavior of a task queue. Learn more about configuring task queue functions at TODO.
```
Comment (davideast) [inline quickstarts/taskqueues-backup-images/README.md:18]:
```suggestion
Task queue functions come with a powerful set of configuration to precisely control rate limits and retry behavior of a task queue. [See the documentation](https://cloud.google.com/tasks/docs/creating-queues) to learn more about configuring task queue functions.
```

**Review by davideast — APPROVED** [review]
I added a few links for "TODO" documentation. Feel free to use something else though.

This looks really good. The only comment I'd have is making some of the code more generic such as throwing `HttpsError`s and whatnot. That's always a fine line when creating samples though. Looks good as is.

File: 2nd-gen/taskqueues-backup-images/functions/index.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        throw new HttpsError("internal", "Uh-oh. Something broke.");
+      }
+    });
+
+
+let auth;
+/**
+ * Returns URL of the given v2 google cloud function.
+ */
+async function getFunctionUrl(name, location="us-central1") {
```
Comment (egilmorez) [inline 2nd-gen/taskqueues-backup-images/functions/index.js:104]: I see from the draft that we should tag this as well.  Thanks!

**Review by egilmorez — COMMENTED** [review]
WDYT Daniel -- time to add the tags for this?  Is it stable at this point?  Thanks!

File: quickstarts/taskqueues-backup-images/README.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
+# Quickstart: Back up images using a Task Queue Function
+This quickstart demonstrates how to setup a Task Queue Function using the Firebase SDK for Cloud Functions.
+
+## Introduction
+
+Task queue functions make it easy to manage the execution, dispatch, and delivery of a large number of distributed tasks.
+
+We leverage its power here to set up a service that backs up all images from NASA's [Astronomy Picture of the Day](https://apod.nasa.gov/apod/astropix.html).
+
+Task queue functions are powered by [Google Cloud Tasks](https://cloud.google.com/tasks). Learn more about the task queue functions at TODO.
```
[the remark this answers is not in the document]
Comment (egilmorez) [reply quickstarts/taskqueues-backup-images/README.md:10]: Suggest our Firebase-ish tutorial, which will live at https://firebase.google.com/docs/functions/beta/task-functions

Nothing there now, but once this sample is tagged, a doc CL will take flight based on the very complete Gdoc draft.

[the remark this answers is not in the document]
Comment (egilmorez) [reply quickstarts/taskqueues-backup-images/README.md:10]: Sorry, for v1 that will be https://firebase.google.com/docs/functions/task-functions

File: 2nd-gen/taskqueues-backup-images/README.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
+# Back up images using a Task Queue Function (v2)
+This quickstart demonstrates how to setup a Task Queue Function using the Firebase SDK for Cloud Functions (v2).
+
+## Introduction
+
+Task queue functions make it easy to manage the execution, dispatch, and delivery of a large number of distributed tasks.
+
+We leverage its power here to set up a service that backs up all images from NASA's [Astronomy Picture of the Day](https://apod.nasa.gov/apod/astropix.html).
+
+Task queue functions are powered by [Google Cloud Tasks](https://cloud.google.com/tasks). Learn more about the task queue functions at TODO.
```
[the remark this answers is not in the document]
Comment (egilmorez) [reply 2nd-gen/taskqueues-backup-images/README.md:10]: Suggest https://firebase.google.com/docs/functions/beta/task-functions

[the remark this answers is not in the document]
Comment (taeold) [reply 2nd-gen/taskqueues-backup-images/README.md]: Going to link to Firebase Docs instead!

File: quickstarts/taskqueues-backup-images/functions/index.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+          stream.on("error", reject);
+        });
+      } catch (err) {
+        logger.error(`Failed to upload ${picUrl} to ${dest.name}`, err);
+        throw new HttpsError("internal", "Uh-oh. Something broke.");
+      }
+      return;
+    });
+
+exports.enqueueBackupTasks = functions.https.onRequest(
```
[the remark this answers is not in the document]
Comment (taeold) [reply quickstarts/taskqueues-backup-images/functions/index.js:100]: Done

File: quickstarts/taskqueues-backup-images/functions/index.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+const BACKUP_START_DATE = new Date("1995-06-17");
+const BACKUP_COUNT = process.env.BACKUP_COUNT || 100;
+const HOURLY_BATCH_SIZE = process.env.HOURLY_BATCH_SIZE || 500;
+const BACKUP_BUCKET = process.env.BACKUP_BUCKET;
+
+/**
+ * Grabs Astronomy Photo of the Day (APOD) using NASA's API.
+ *
+ */
+exports.backupApod = functions
```
[the remark this answers is not in the document]
Comment (taeold) [reply quickstarts/taskqueues-backup-images/functions/index.js:36]: Gave a first go!

**Review by egilmorez — APPROVED** [review]
LG, thanks!

**Pull request merged**